src/tests/tests.spec.js

```diff
@@ -0,0 +1,91 @@
+    for (const value of filterValues) {
+        test(`Perform and verify sorting on the Inventory page ${value}`, async ({ inventoryPage }) => {
+            const defaultList = await inventoryPage.getInventoryItemsList();
+            const sortedDefaultList = performDefaultListSorting(defaultList, value);
+            await inventoryPage.performSorting(value);
+            const sortedList = await inventoryPage.getInventoryItemsList();
+            expect(sortedList).toEqual(sortedDefaultList);
+        });
+    }
+
+    test('Verification of Display Accuracy after Adding Random Products to Cart', async ({ inventoryPage, shoppingCartPage }) => {
+        const allInventoryItemsList = await inventoryPage.getInventoryItemsList();
+        const randomItemsIndexesArray = await inventoryPage.getRandomItemsIndexes();
+        await inventoryPage.addRandomItemsToCart(randomItemsIndexesArray);
+        await inventoryPage.shoppingCart.click();
+        const allCartItemsList = await shoppingCartPage.getCartItemsList();
+        expect(allInventoryItemsList).toEqual(expect.arrayContaining(allCartItemsList));
+    });
+
+    test('Verification of Display Accuracy after Adding Random Products to Cart v2', async ({ inventoryPage, shoppingCartPage }) => {
+        const allInventoryItemsList = await inventoryPage.getInventoryItemsList();
+        const randomItemsIndexesArray = await inventoryPage.getRandomItemsIndexes();
+        // const selectedItemsData = await inventoryPage.getRandomItemsData(randomItemsIndexesArray);
+        await inventoryPage.addRandomItemsToCart(randomItemsIndexesArray);
+        await inventoryPage.shoppingCart.click();
+        const cartItemsListData = await shoppingCartPage.getCartItemsList();
+        // тут виходить, що я кожну проперті доданого айтему, порівнюю з поперті айтему масиву усих елементів, тому наче має бути норм
+        for (const i of randomItemsIndexesArray) {
+            const k = randomItemsIndexesArray.indexOf(i);
+            expect(await allInventoryItemsList[i].name).toEqual(cartItemsListData[k].name);
+            expect(await allInventoryItemsList[i].description).toEqual(cartItemsListData[k].description);
+            expect(await allInventoryItemsList[i].price).toEqual(cartItemsListData[k].price);
+        }
+    });
+
+    test('Proceeding to Checkout, and verifying product details and Total Price accuracy', async ({
+        inventoryPage, shoppingCartPage, checkoutStepOnePage, checkoutStepTwoPage,
+    }) => {
+        const inventoryItemsListData = await inventoryPage.getInventoryItemsList();
+        const randomItemsIndexesArray = await inventoryPage.getRandomItemsIndexes();
+        await inventoryPage.addRandomItemsToCart(randomItemsIndexesArray);
+        await inventoryPage.shoppingCart.click();
+        await shoppingCartPage.checkoutButton.click();
+        await checkoutStepOnePage.fillAndConfirmCheckoutStepOne('Ssssss', 'Sssssss', '45674');
```

Comment:
user data better to store in some variable

Reply:
Moved them to the parameters.